internal/providers/amazon/eks/driver/cluster_creator.go

```diff
@@ -0,0 +1,511 @@
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+
+	"github.com/banzaicloud/pipeline/internal/cluster/metrics"
+	"github.com/banzaicloud/pipeline/internal/secret/ssh"
+	"github.com/banzaicloud/pipeline/internal/secret/ssh/sshdriver"
+	"github.com/banzaicloud/pipeline/src/auth"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/sirupsen/logrus"
+	logrusadapter "logur.dev/adapter/logrus"
+
+	"github.com/banzaicloud/pipeline/internal/cloudinfo"
+	"github.com/banzaicloud/pipeline/src/secret/verify"
+
+	"go.uber.org/cadence/client"
+
+	"github.com/banzaicloud/pipeline/internal/providers/amazon/eks/workflow"
+	pkgCluster "github.com/banzaicloud/pipeline/pkg/cluster"
+	pkgEks "github.com/banzaicloud/pipeline/pkg/cluster/eks"
+	pkgErrors "github.com/banzaicloud/pipeline/pkg/errors"
+	pkgEC2 "github.com/banzaicloud/pipeline/pkg/providers/amazon/ec2"
+	"github.com/banzaicloud/pipeline/src/cluster"
+)
+
+type EksClusterCreator struct {
+	logger                     logrus.FieldLogger
+	workflowClient             client.Client
+	cloudInfoClient            *cloudinfo.Client
+	clusters                   clusterRepository
+	secrets                    secretValidator
+	statusChangeDurationMetric metrics.ClusterStatusChangeDurationMetric
+	clusterTotalMetric         *prometheus.CounterVec
+}
+
+type secretValidator interface {
+	ValidateSecretType(organizationID uint, secretID string, cloud string) error
+}
+
+type clusterRepository interface {
+	Exists(organizationID uint, name string) (bool, error)
+}
+
+func NewEksClusterCreator(logger logrus.FieldLogger,
+	workflowClient client.Client,
+	cloudInfoClient *cloudinfo.Client,
+	clusters clusterRepository,
+	secrets secretValidator,
+	statusChangeDurationMetric metrics.ClusterStatusChangeDurationMetric,
+	clusterTotalMetric *prometheus.CounterVec) EksClusterCreator {
+	return EksClusterCreator{
+		logger:                     logger,
+		workflowClient:             workflowClient,
+		cloudInfoClient:            cloudInfoClient,
+		clusters:                   clusters,
+		secrets:                    secrets,
+		statusChangeDurationMetric: statusChangeDurationMetric,
+		clusterTotalMetric:         clusterTotalMetric,
+	}
+}
+
+func getNodePoolsForSubnet(subnetMapping map[string][]*pkgEks.Subnet, eksSubnet workflow.Subnet) []string {
+	var nodePools []string
+	for np, subnets := range subnetMapping {
+		for _, subnet := range subnets {
+			if (subnet.SubnetId != "" && eksSubnet.SubnetID == subnet.SubnetId) ||
+				(subnet.Cidr != "" && eksSubnet.Cidr == subnet.Cidr) {
+				nodePools = append(nodePools, np)
+			}
+		}
+	}
+	return nodePools
+}
+
+// Create implements the clusterCreator interface.
+func (c *EksClusterCreator) create(ctx context.Context, commonCluster cluster.CommonCluster, createRequest *pkgCluster.CreateClusterRequest) (cluster.CommonCluster, error) {
+	c.logger.Info("start creating EKS Cluster")
```

Comment:
Add context to the logger such as `cluster` , `organizationID`, `correlationID` etc